modules/imonomyBidAdapter.js

```diff
@@ -0,0 +1,258 @@
+      }
+
+      $$PREBID_GLOBAL$$[callbackName] = handleCallback(bids);
+
+      adloader.loadScript(request.join('&'));
+    } catch (e) {
+      return;
+    }
+  }
+
+  function formRequestUrl(reqData) {
+    var key;
+    var url = [];
+
+    for (key in reqData) {
+      if (reqData.hasOwnProperty(key) && reqData[key])
+        url.push(key, '=', reqData[key], '&');
+    }
+
+    return encode64(url.join('').slice(0, -1));
+  }
+
+  function handleCallback(bids) {
+    return function handleResponse(adItems) {
+      try {
+        var bidObject;
+        var bidder = 'imonomy';
+        var adItem;
+        var bid;
+
+        try{
+          if(adItems.um_list){
+            _processUserMatchings(adItems.um_list);
+          }
+        } catch(e) {}
+
+        try{
+          if (adItems.um_hook){eval(adItems.um_hook);}
+        }catch(e){}
+
+        adItems = adItems.ads;
+
+        for (var i = 0, l = bids.length; i < l; i++) {
+          bid = bids[i];
+          adItem = getAdItem(adItems, bids[i].bidId);
+
+          if (adItem) {
+            try{
+              if (adItem.settings){eval(adItem.settings);}
+            } catch(e) {}
+            bidObject = bidfactory.createBid(STATUSCODES.GOOD, bid);
+            bidObject.bidderCode = bidder;
+            bidObject.ad         = adItem.ad;
+            bidObject.cpm        = adItem.cpm;
+            bidObject.cur        = adItem.cur;
+            bidObject.width      = adItem.width;
+            bidObject.height     = adItem.height;
+            bidmanager.addBidResponse(bid.placementCode, bidObject);
+          } else {
+            bidObject = bidfactory.createBid(STATUSCODES.NO_BID, bid);
+            bidObject.bidderCode = bidder;
+            bidmanager.addBidResponse(bid.placementCode, bidObject);
+          }
+
+        }
+      } catch (e) {
+        return;
+      }
+    };
+  }
+
+  function _processUserMatchings(userMatchings) {
+    var headElem = document.getElementsByTagName('head')[0],
+      createdElem;
+
+    utils._each(userMatchings, function (userMatching) {
+      createdElem = undefined;
+      switch (userMatching.Type) {
+        case 'redirect':
+          createdElem = document.createElement('img');
+          break;
+        case 'iframe':
+          createdElem = utils.createInvisibleIframe();
+          break;
+        case 'js':
+          createdElem = document.createElement('script');
+          createdElem.type = 'text/javascript';
+          createdElem.async = true;
+          break;
+      }
+      if (createdElem) {
+        createdElem.src = decodeURIComponent(userMatching.Url);
+        headElem.insertBefore(createdElem, headElem.firstChild);
+      }
+    });
+  }
+
+  function getAdItem(adItems, imp){
+    adItems = adItems || []
+    for (var i = 0, l = adItems.length; i < l; i++) {
+      if(adItems[i].impression_id == imp ){
+        return adItems[i];
+      }
+    }
+
+    return null;
+  }
+
+  function encode64(input) {
```

Comment:
Are you just re-implementing [atob](https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding) here? Or am I missing something?